src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/CSharpOrVisualBasicFileRenameHandler.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+//--------------------------------------------------------------------------------------------
+// <summary>
+// CSharpOrVisualBasicFileRenameHandler
+//
+// Exports an IFileRenameHandler to listen to handle renames. If the file being renamed
+// is a code file, it will prompt the user to rename the class to match. The rename is done
+// using Roslyn Renamer API
+// </summary>
+//--------------------------------------------------------------------------------------------
+using System;
+using System.ComponentModel.Composition;
+using System.Globalization;
+using System.IO;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.ProjectSystem.Rename;
+using Microsoft.VisualStudio.ProjectSystem.Waiting;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Rename
+{
+    [Export(typeof(IFileRenameHandler))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
```

Comment:
What exactly makes this C#/VB specific but not RenameTypeService?

Reply:
That is a good question.  I assume that F# does not implement code model but I will check on this.